commit cf5d99eeefb53fef4a9a02ad078c6fda8b81e724
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2017-10-26T21:04:19+00:00 (1 year, 11 months ago)
    
    MANTA-2412 Allow addition of content-disposition header to manta objects

diff --git a/lib/moray.js b/lib/moray.js
index ca89840..1097e66 100644
--- a/lib/moray.js
+++ b/lib/moray.js
@@ -337,6 +337,8 @@ function createMetadata(options) {
 
     switch (options.type) {
     case 'object':
+        assert.optionalString(options.contentDisposition,
+                'options.contentDisposition');
         assert.number(options.contentLength, 'options.contentLength');
         assert.string(options.contentMD5, 'options.contentMD5');
         assert.string(options.contentType, 'options.contentType');
@@ -350,6 +352,7 @@ function createMetadata(options) {
             });
         }
 
+        md.contentDisposition = options.contentDisposition;
         md.contentLength = options.contentLength;
         md.contentMD5 = options.contentMD5;
         md.contentType = options.contentType;
